File: 2-Android With Java/MVC/Day5_Threads/app/src/main/java/com/example/day5_threads/mainscreen/controller/MainActivity.java
package com.example.day5_threads.mainscreen.controller;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

import com.example.day5_threads.R;
import com.example.day5_threads.allproducts.controller.MainMenuActivity;
import com.example.day5_threads.favproducts.controller.FavActivity;

// MainMenuActivity.java (for your menu with Main/Fav/Exit buttons)
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main);

        Button btnMain = findViewById(R.id.Main);
        Button btnFav = findViewById(R.id.Fav);
        Button btnExit = findViewById(R.id.button3);

        btnMain.setOnClickListener(v -> {
            startActivity(new Intent(this, MainMenuActivity.class));
        });

        btnFav.setOnClickListener(v -> {
            startActivity(new Intent(this, FavActivity.class));
        });

        btnExit.setOnClickListener(v -> {
            finish();
        });
    }
}

File: 3-Android with kotlin/Day6/ViewBind/app/src/main/java/com/example/day1/activity1/view/MainActivity.kt
package com.example.day1.activity1.view

import android.content.Intent
import android.content.res.Configuration.ORIENTATION_PORTRAIT
import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentTransaction
import com.example.day1.R
import com.example.day1.activity2.view.Fragment2
import com.example.day1.activity2.view.MainActivity2
import com.example.day1.model.remote.Product
//import com.example.day1.work.ProductWork

class MainActivity : AppCompatActivity() , Communicator
{
    private lateinit var dynamicFragment: ProductListFragment
    private lateinit var fragmentMngr :FragmentManager
    private lateinit var fragmentTransaction:FragmentTransaction

    private var dynamicFragment2: Fragment2? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)



        dynamicFragment = ProductListFragment()
        fragmentMngr = supportFragmentManager
        fragmentTransaction = supportFragmentManager.beginTransaction()
        fragmentTransaction.replace(R.id.staticFragmentContainer , dynamicFragment , "activity1")
        fragmentTransaction.commit()

    }

    override fun ShowProductDetails(it: Product)
    {
        if( resources.configuration.orientation== ORIENTATION_PORTRAIT)
        {

            val intent = Intent(this , MainActivity2::class.java)
            intent.putExtra("product" , it)
            startActivity(intent)
        }
        else
        {
            if (dynamicFragment2 == null)
            {
                dynamicFragment2= Fragment2()
                supportFragmentManager.beginTransaction()
                    .replace(R.id.FragmentContainer, dynamicFragment2!!, "fragment2")
                    .commit()
                supportFragmentManager.executePendingTransactions()

            }
            dynamicFragment2?.ShowProduct(it)

        }
    }
}